controllers/flinkcluster_reconciler.go

```diff
@@ -210,28 +208,33 @@ func (reconciler *_ClusterReconciler) deleteService(
 	return err
 }
 
-func (reconciler *_ClusterReconciler) reconcileJob() error {
+func (reconciler *_ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 	var log = reconciler.log
 	var desiredJob = reconciler.desiredState.Job
-	var observedJob = reconciler.observedState.job
-	var observedClusterComponents = reconciler.observedState.cluster.Status.Components
+	var observed = reconciler.observedState
+	var observedJob = observed.job
+
 	if desiredJob != nil {
 		if observedJob == nil {
-			var jmDeploymentReady = observedClusterComponents.JobManagerDeployment.State ==
-				flinkoperatorv1alpha1.ClusterComponentState.Ready
-			var jmServiceReady = observedClusterComponents.JobManagerService.State ==
-				flinkoperatorv1alpha1.ClusterComponentState.Ready
-			var tmDeploymentReady = observedClusterComponents.TaskManagerDeployment.State ==
-				flinkoperatorv1alpha1.ClusterComponentState.Ready
-			if jmDeploymentReady && jmServiceReady && tmDeploymentReady {
-				return reconciler.createJob(desiredJob)
+			// If the observed Flink job status list is not nil (e.g., emtpy list), it means
+			// Flink REST API server is up and running. It is the source of truth of whether
+			// we can submit a job.
+			if reconciler.observedState.flinkJobStatusList != nil {
```

Comment:
just observed? here and everywhere in this method.

Reply:
I don't understand your question. If the observed flinkJobStatusList is not nil, it means we have call Flink API successfully, then we can submit job to Flink.